conwayste/src/uimanager.rs

```diff
@@ -0,0 +1,94 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of conwayste.
+ *
+ *  conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with conwayste.  If not, see
+ *  <http://www.gnu.org/licenses/>. */
+#![allow(unused)]
+
+use crate::Screen;
+use crate::uilayout::UILayout;
+use crate::ui::{
+    Button,
+    Chatbox,
+    Checkbox,
+    Label,
+    Layer,
+    Pane,
+    TextField,
+    WidgetID,
+};
+
+// When adding support for a new widget, use this macro to define a routine to search in the
+// `UILayout` under the `Screen` for a `WidgetID` of that type
+macro_rules! widget_from_screen_and_ui {
```

Comment:
This name is very similar to `widget_from_screen_and_id`. I'd kind of like a different name.